### ISSUE TITLE
reproduction error

### ISSUE THREAD

**Issue by Meiling510** [ShuGuoJ/3DAES#6]
I use the logistic regression classifier(LogisticRegression(multi_class="multinomial", solver="newton-cg", max_iter=1000)) for classification, which cannot achieve the effect of the paper. Would you like to share the source code of the third part for reproduction?

**Comment by ShuGuoJ** [comment]
In my code, the logistic regression implemented in sklearn (a third-party library) is used as the classifier. Specifically, `classifier = LogisticRegression(random_state=666, max_iter=4000)`. The reason caused the bad result compared with ours may be: 1. a good feature model is not selected; 2. the distribution of training samples is not very well, which would degrade the model, especially when there are just a few available samples for training. Hope my response is useful for you.

**Comment by Meiling510** [comment]
Thank you for your reply! I'll try again.

**Comment by Meiling510** [comment]
I tried it again, but the result is still not good.After I train AE,I chose the 5th epoch feature model which has the smallest test error,and then trained Siamese network for 10 epoch with 10 samples per class also as in your paper,and finally fit a Logistic Regression classifier with the same 10 samples per class in training siamese.I used PaviaU to train the model,but get an OA of 0.5. (all elements in confusion matrix: [0.         0.80221895 0.20020222 0.54737733 0.         0.13327427
 0.27259368 0.30137224 0.92977528]).And i used trainTestSplit.py to randomly pick 10 samples per class as training data as same as yours.So i think there is no uneven distribution of training samples.Do you know where my problem is？

**Comment by ShuGuoJ** [comment]
I think it is very abnormal. When I only use autoencoder to extract features, the classification accuracy has outperformed 50%. I suggest you can check whether label indexes in training correspond to ones in inference.

**Comment by Meiling510** [comment]
I can't find the YRE dataset, could you be generous enough to share it？

**Comment by ShuGuoJ** [comment]
I am sorry to reject it for the copyright of YRE is private and does not belong to us.




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年9月21日(星期三) 晚上7:55
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [ShuGuoJ/3DAES] reproduction error (Issue #6)





 
I can't find the YRE dataset, could you be generous enough to share it？
